homeassistant/components/light/zwave.py

```diff
@@ -107,3 +142,178 @@ def turn_off(self, **kwargs):
         """Turn the device off."""
         if self._value.node.set_dimmer(self._value.value_id, 0):
             self._state = STATE_OFF
+
+
+def ct_to_rgb(temp):
+    """Convert color temperature (mireds) to RGB."""
+    colorlist = list(
+        color_temperature_to_rgb(color_temperature_mired_to_kelvin(temp)))
+    return [int(val) for val in colorlist]
+
+
+class ZwaveColorLight(ZwaveDimmer):
+    """Representation of a Z-Wave color changing light."""
+
+    def __init__(self, value):
+        """Initialize the light."""
+        self._value_color = None
+        self._value_color_channels = None
+        self._color_channels = None
+        self._rgb = None
+        self._ct = None
+
+        # Here we attempt to find a zwave color value with the same instance
+        # id as the dimmer value. Currently zwave nodes that change colors
+        # only include one dimmer and one color command, but this will
+        # hopefully provide some forward compatibility for new devices that
+        # have multiple color changing elements.
+        for value_color in value.node.get_rgbbulbs().values():
+            if value.instance == value_color.instance:
+                self._value_color = value_color
+
+        if self._value_color is None:
+            raise ValueError("No matching color command found.")
+
+        for value_color_channels in value.node.get_values(
+                class_id=zwave.COMMAND_CLASS_COLOR, genre='System',
+                type="Int").values():
+            self._value_color_channels = value_color_channels
+
+        if self._value_color_channels is None:
+            raise ValueError("Color Channels not found.")
+
+        super().__init__(value)
+
+    @property
+    def name(self):
+        """Return the name of the device."""
+        # In the case of a color changing light, use only the node name to
+        # generate the friendly device name. Currently color changing bulbs
+        # have a one node to one entity relationship, and the entity will
+        # include both the dimmer and the color zwave value. This allows a
+        # friendlier name assignment in this case.
+        return zwave.node_name(self._value.node)
```

Comment:
We should not overwrite this method. I prefer consistency in naming (with the other Z-Wave devices) over the chance that a person would not want to rename the device as it is currently named. Chances are probably still small as people tend to name their lights to the location in the house instead of the vendor.